src/test/java/com/microsoft/sqlserver/jdbc/resultset/ResultSetTest.java

```diff
@@ -510,4 +511,89 @@ public void testResultSetMethods() throws SQLException {
             }
         }
     }
+
+    @Test
+    public void testMultipleResultSets() throws SQLException {
+        try (Connection con = getConnection(); Statement stmt = con.createStatement()) {
+            stmt.execute(
+                    "CREATE TABLE " + AbstractSQLGenerator.escapeIdentifier(tableName) + " (c1 int IDENTITY, c2 int)");
+            String SQL = "exec sp_help 'dbo." + TestUtils.escapeSingleQuotes(tableName) + "'";
+
+            boolean results = stmt.execute(SQL);
+            int rsCount = 0;
+
+            // Loop through the available result sets.
+            while (results) {
+                try (ResultSet rs = stmt.getResultSet()) {
+                    rsCount++;
+                    String firstColumnName = rs.getMetaData().getColumnName(1);
+                    switch (rsCount) {
+                        case 1:
+                            assert (firstColumnName.trim().equalsIgnoreCase("Name"));
+                            break;
+                        case 2:
+                            assert (firstColumnName.trim().equalsIgnoreCase("Column_name"));
+                            break;
+                        case 3:
+                            assert (firstColumnName.trim().equalsIgnoreCase("Identity"));
+                            break;
+                        case 4:
+                            assert (firstColumnName.trim().equalsIgnoreCase("RowGuidCol"));
+                            break;
+                        case 5:
+                            assert (firstColumnName.trim().equalsIgnoreCase("Data_located_on_filegroup"));
+                            break;
+                    }
+                    int i = 1;
+                    while (rs.next()) {
+                        String firstColumnValue = rs.getString(1);
+                        switch (rsCount) {
+                            case 1:
+                                assert (firstColumnValue.equals(tableName));
+                                break;
+                            case 2:
+                                assert (firstColumnValue.equalsIgnoreCase("c" + i++));
+                                break;
+                            case 3:
+                                assert (firstColumnValue.equalsIgnoreCase("c1"));
+                                break;
+                            case 4:
+                                assert (firstColumnValue.equalsIgnoreCase("No rowguidcol column defined."));
+                                break;
+                            case 5:
+                                assert (firstColumnValue.equalsIgnoreCase("PRIMARY"));
+                                break;
+                        }
+                    }
+                }
+
+                SQLWarning warnings = stmt.getWarnings();
```

Comment:
Shouldn't we also validate the warnings?

Reply:
These warnings are not consistent and I don't think there is a clean way of testing them. Sometimes they're just blank lines. We just care about letting them pass through.

Reply:
Can we at least validate that warnings exist (not `null`)?

Reply:
Added a validation for their count (hardcoded value is based on what database responds with). We can confirm with this that all the warnings were captured.